src/utilstrencodings.cpp

```diff
@@ -99,10 +100,12 @@ bool IsHex(const string& str)
     return (str.size() > 0) && (str.size()%2 == 0);
 }
 
-vector<unsigned char> ParseHex(const char* psz)
+v_uint8 ParseHex(const char* psz)
 {
     // convert hex dump to vector
-    vector<unsigned char> vch;
+    v_uint8 vch;
+    const size_t nLength = psz ? strlen(psz) : 0;
```

Comment:
vch has to be returned in any case, if nLength is 0 or not.
So why we need to add any extra checks here?
The point is - if code works in a special case (like this when nLength=0) without any problems, then why to introduce any extra checks? To reduce performance?

Reply:
I've corrected ParseHex here only to improve performance -> to make reservation for the vector. Without that it was simply reallocating memory for every new char.

Reply:
I thought to add return to exclude moving to next code lines (reserve, while{})

Reply:
we need to avoid multiple return statements in the code if possible.
This is a best coding practice. Look at https://en.wikipedia.org/wiki/Return_statement (Multiple return statements section)...
"The most common problem in early exit is that cleanup or final statements are not executed – for example, allocated memory is not unallocated, or open files are not closed, causing leaks."
It's not that important in this case, but use it as a rule - good practice.

Reply:
ok. will follow